lib/jsdom/living/xhr/XMLHttpRequest-impl.js

```diff
@@ -83,945 +80,883 @@ const corsSafeResponseHeaders = new Set([
-        throw new TypeError("2 arguments required for setRequestHeader");
-      }
-      header = conversions.ByteString(header);
-      value = conversions.ByteString(value);
+    if (this.readyState !== XMLHttpRequestImpl.OPENED || properties.send) {
+      throw DOMException.create(this._globalObject, ["The object is in an invalid state.", "InvalidStateError"]);
+    }
 
-      if (this.readyState !== XMLHttpRequest.OPENED || properties.send) {
-        throw DOMException.create(this._globalObject, ["The object is in an invalid state.", "InvalidStateError"]);
-      }
+    value = normalizeHeaderValue(value);
 
-      value = normalizeHeaderValue(value);
+    if (!tokenRegexp.test(header) || !fieldValueRegexp.test(value)) {
+      throw DOMException.create(this._globalObject, [
+        "The string did not match the expected pattern.",
+        "SyntaxError"
+      ]);
+    }
 
-      if (!tokenRegexp.test(header) || !fieldValueRegexp.test(value)) {
-        throw DOMException.create(this._globalObject, [
-          "The string did not match the expected pattern.",
-          "SyntaxError"
-        ]);
-      }
+    const lcHeader = header.toLowerCase();
 
-      const lcHeader = header.toLowerCase();
+    if (forbiddenRequestHeaders.has(lcHeader) || lcHeader.startsWith("sec-") || lcHeader.startsWith("proxy-")) {
+      return;
+    }
 
-      if (forbiddenRequestHeaders.has(lcHeader) || lcHeader.startsWith("sec-") || lcHeader.startsWith("proxy-")) {
+    const keys = Object.keys(flag.requestHeaders);
+    let n = keys.length;
+    while (n--) {
+      const key = keys[n];
+      if (key.toLowerCase() === lcHeader) {
+        flag.requestHeaders[key] += ", " + value;
         return;
       }
-
-      const keys = Object.keys(flag.requestHeaders);
-      let n = keys.length;
-      while (n--) {
-        const key = keys[n];
-        if (key.toLowerCase() === lcHeader) {
-          flag.requestHeaders[key] += ", " + value;
-          return;
-        }
-      }
-      flag.requestHeaders[header] = value;
-    }
-
-    get _ownerDocument() {
-      return idlUtils.implForWrapper(window.document);
     }
+    flag.requestHeaders[header] = value;
   }
+}
 
-  Object.defineProperty(XMLHttpRequest.prototype, Symbol.toStringTag, {
-    value: "XMLHttpRequest",
-    writable: false,
-    enumerable: false,
-    configurable: true
-  });
+setupForSimpleEventAccessors(XMLHttpRequestImpl.prototype, ["readystatechange"]);
 
-  setupForSimpleEventAccessors(XMLHttpRequest.prototype, ["readystatechange"]);
+Object.defineProperty(XMLHttpRequestImpl, "UNSENT", { value: 0, enumerable: true });
+Object.defineProperty(XMLHttpRequestImpl, "OPENED", { value: 1, enumerable: true });
+Object.defineProperty(XMLHttpRequestImpl, "HEADERS_RECEIVED", { value: 2, enumerable: true });
+Object.defineProperty(XMLHttpRequestImpl, "LOADING", { value: 3, enumerable: true });
+Object.defineProperty(XMLHttpRequestImpl, "DONE", { value: 4, enumerable: true });
```

Comment:
Can we just use simple assignment? I don't think the writability/configurability settings have too many benefits (as in I don't think anyone would do `XMLHttpRequestImpl.UNSENT = …` in their code).

```js
// Make constants available on both the class and its instances, like in Web IDL.
for (const obj of [XMLHttpRequestImpl, XMLHttpRequestImpl.prototype]) {
  obj.UNSENT = 0;
  // …
}
```